sdk/python/kfp/compiler/v2_compatible_compiler_test.py

```diff
@@ -90,8 +88,28 @@ def v2_compatible_two_step_pipeline():
           num_steps=preprocess_task.outputs['output_parameter_one'],
           dataset=preprocess_task.outputs['output_dataset_one'])
 
+    kfp_compiler = compiler.Compiler(
+        mode=dsl.PipelineExecutionMode.V2_COMPATIBLE)
+    self._assert_compiled_pipeline_equals_golden(
+        kfp_compiler, v2_compatible_two_step_pipeline,
+        'v2_compatible_two_step_pipeline.yaml')
+
+  def test_custom_launcher(self):
+
+    @dsl.pipeline(pipeline_root='gs://output-directory/v2-artifacts',
+                  name='my-test-pipeline-with-custom-launcher')
+    def v2_compatible_two_step_pipeline():
+      preprocess_task = preprocess_op(uri='uri-to-import', some_int=12)
+      train_task = train_op(
+          num_steps=preprocess_task.outputs['output_parameter_one'],
+          dataset=preprocess_task.outputs['output_dataset_one'])
+
+    kfp_compiler = compiler.Compiler(
+        mode=dsl.PipelineExecutionMode.V2_COMPATIBLE,
+        launcher_image='my-custom-image')
     self._assert_compiled_pipeline_equals_golden(
-        v2_compatible_two_step_pipeline, 'v2_compatible_two_step_pipeline.yaml')
+        kfp_compiler, v2_compatible_two_step_pipeline,
+        'v2_compatible_two_step_pipeline_with_custom_launcher.yaml')
```

Comment:
JFYI: Such tests (based on golden compiled pipelines) are really fragile and add quite a lot of maintenance cost. They will start failing if any small thing is changed in the code generator or pipeline compiler.
We advice writing test cases that only check the required parts of the compiled pipeline. Namely the launcher image used in the templates.

Reply:
Both kinds of tests seem reasonable to me.